### ISSUE TITLE
macOS Mojave - Extend ToolKit and PPRO 13 - Impossible to use

### ISSUE THREAD

**Issue by MacyEleanorMay** [Adobe-CEP/Samples#100]
In case you did not know ExtenScript Toolkit doesn't work on Mojave.
Entering in one line of code takes 40 SWODs and 60 seconds. 

No one can work like that

**Comment by bbb999** [comment]
ESTK works fine on Mojave.

https://medium.com/adobetech/workaround-for-extendscript-toolkit-debugger-error-1116-f067f81f96c6

**Comment by dokluch** [comment]
@bbb999 here's a short screencast on this issue:
https://www.youtube.com/watch?v=VR-cffGnCpY

**Comment by bbb999** [comment]
That looks frustrating. I'm not sure what's going on in that video.

Many of us run ESTK every day on Mojave, without those issues; Mojave is not the sole cause of that problem.

**Comment by dokluch** [comment]
@bbb999 well as you see, it's:
1. Not only problem of mine
2. I've tested this on 3 MacOS devices I own. All the same. Even a clean install.

So this doesn't help at all, and no, ESTK is not working and it seems that you are just lucky to have it up and working. I've been too disappointed with this tool a long time ago and truly have neither time nor will to try and fix it. It's just not working and I have to find weird workarounds.

**Comment by bbb999** [comment]
>  it seems that you are just lucky to have it up and working. 

I'm responsible for all integration with Adobe's video and audio products, including hundreds of panels, reliant on ExtendScript. So far, no partners have reported the issues you're having. I don't mean to imply that you (and others) aren't running into a very annoying issue; just that it's possible for many developers to use ESTK on current MacOS, without hitting the issue.

More info coming soon, from the Extensibility team.

**Comment by dokluch** [comment]
I do understand that and I highly appreciate your job. 
It does seem like a very strange and highly specific issue. I've been trying to remove as much apps as possible from my system and make a clean reinstall but this issue always gets back to me and I'm effectively cut off from efficient ESTK development and that annoys me a lot.

**Comment by bbb999** [comment]
Agreed, the behavior looks very frustrating. 

Confirming; you see that with every app to which you connect, and not just AE, right?

**Comment by jondrometa86** [comment]
@dokluch - I can confirm what you're experiencing is normal for me too. 

I've also worked from multiple Mac computers (iMac 2014, MacBook Air 2016, and MacBook Pro 2018) using Mojave (10.13) and started to believe this was just the standard behavior on Mac OS... but no one liked to talk about it.

### **Here are some solutions I've adopted to circumvent the lag issue:**

1. **Toggle the "Prevent App Nap" option from ESTK's "Get Info panel".**
Access this by selecting the ESTK app in Finder. Press [CMD+i] to pull up the "Info" panel and tick the "Prevent App Nap" box. 
_Note- This worked for about 2 weeks before it got weird again._

2. **Try making the AE window visible while using ESTK.**
For some reason, having the windows side by side seems to help ESTK with lag issues on my system.

3. **If ESTK gets stuck lagging uncontrollably, try hitting the [ESC] key multiple times.**
If this doesn't work... Click on the AE window (or toggle over using CMD+Tab), then click back on the ESTK window. 


### Going forward, I highly advise moving to a different script editor that has an Adobe Script Running function:
- [Atom](https://atom.io/) has [Adobe Script Runner](https://atom.io/packages/adobe-script-runner)
- [Brackets](http://brackets.io/) (ironically, by Adobe) has "[Brackets to Photoshop](https://github.com/JavierAroche/brackets-to-photoshop)" which also works with AE.
- The Adobe Tech Blog recently announced [building an official plugin for VSCode](https://medium.com/adobetech/the-future-of-extendscript-development-a-vscode-plugin-2d8d0172a357)

Any of these IDE's are 1000x better for writing your JSX scripts. They're modern, no lag time, and the software isn't stuck in 2005 (no offense, @bbb999, just keeping it real).

If your script hits an issues while running, ESTK will pop up for debugging. That's when you use the aforementioned tips to avoid lag time.

**In short:** This is a very real problem I've had to deal with while learning Javascript in ESTK. I forced myself to find ways around it. 

Hope this helps you and anyone else suffering with the issue!

**Comment by MacyEleanorMay** [comment]
> I'm responsible for all integration with Adobe's video and audio products, including hundreds of panels, reliant on ExtendScript. 
> More info coming soon, from the Extensibility team.

@bbb999 

Panels are basically not working on a Mac on PPRO. Just install Mojave or HS on a fresh Mac along with PPRO - then go to CC and download some of the panels. I am sure you'll be pulling your hairs out the head wondering why it is not working...

Also -- Open Premiere Pro press (windows) CTRL+f12 or (Mac) CMD+f12, and type the following into Premiere Console, then relaunch Premiere.... Only thing is that hitting CMD+f12 in PPRO 2019 will NOT open that console.... This is all really really frustrating !

**Comment by bbb999** [comment]
Panels are basically not working on a Mac on PPRO. 

Happily, not so;  panels work great on numerous Mojave systems for me, and all third party developers with whom we work. They're not shy about telling us, when stuff _doesn't_ work. :)

Of course, my-and-their success don't help you; I'd like to get to the bottom of the issue you're seeing...

When you say the panels 'don't work', what specific behavior are you seeing? Do they not appear? Slow response to interactions? Please send any/all specifics, and (if possible) step-by-step instructions on how to repro the behavior, to my day job address: bbb@adobe.com

I have just confirmed that the Console window works in 12.x (2018) and 13.x (2019), here, and haven't heard of any problems opening the Console. Out of curiosity, what do you need to do in the Console window?

**Comment by MacyEleanorMay** [comment]
I am hitting you message via email

**Comment by MacyEleanorMay** [comment]
Installing via Creative Cloud is unsuccessful... 
Installing via Anastasiy does the job...

**Comment by bbb999** [comment]
Yeah, I saw that in the video; none of the extensions you picked from the web page, seem to actually be installed in your /extensions directory. 

Is CC Desktop file syncing ON?

**Comment by ErinFinnegan** [comment]
Just to update you, the ESTK VSCode plugin for Mac is now in prerelease: https://medium.com/adobetech/extendscript-debugger-for-visual-studio-code-mac-prerelease-84040d6cab92

**Comment by bbb999** [comment]
From an email conversation with MacyEleanorMay:

> 
> Found the culprit... It is Universal Audio. IN particular their UA Mixer Engine (one can find it in Activity Monitor) that for the entire macOS prevents user from using CMD+F12.
> Force Quitting it will immediately remedy issue. From User/Groups one can disable it from running at login.
> 
> However, this renders UAAudio Interface non and void and we're using them all over the place..
> 
> Anyway, I have reported this to UA but now you know too -)
